.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,44 @@
+name: benchmark
+permissions: read-all
+
+on:
+  pull_request:
+    branches:
+    paths:
+      - '.github/workflows/benchmark.yaml'
+      - 'benchmark/**'
+  push:
+    branches: [ master ]
+    paths:
+      - '.github/workflows/benchmark.yaml'
+      - 'benchmark/**'
+  schedule:
+    - cron: '0 0 * * 0' # weekly
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: benchmark
+    strategy:
+      matrix:
+        sdk: dev
+    
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - uses: dart-lang/setup-dart@e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c
+        with:
+          sdk: ${{matrix.sdk}}
+
+      - run: dart pub get
+
+      - run: dart analyze --fatal-infos
+
+      - run: dart format --output=none --set-exit-if-changed lib
+
+      - run: ./../tool/setup.sh
+
+      - run: ./tool/compile_protos.sh
+
+      - run: dart test
```

Comment:
`dart test` (after all the preparation steps) fails on my system because there are no tests in this directory.

Instead of `dart test` we should analyze each of the files in `bin/` or maybe even run them.

To compile the programs we have a Dart script in `tool/compile_benchmarks.dart`. I feel like it we just analyze the files in `bin/` or compile them that should be fine. Running will take long and I'm not sure if running them can catch any issues. (after `dart analyze` passes)